ember-core/shared/src/main/scala/org/http4s/ember/core/Encoder.scala

```diff
@@ -121,33 +115,25 @@ private[ember] object Encoder {
               .append(CRLF)
           }
 
-        req.headers
-          .get[`Content-Length`]
-          .foreach { _ =>
-            appliedContentLength = true
-          }
-
         // Apply each header followed by a CRLF
         req.headers.foreach { h =>
           if (h.isNameValid) {
-            stringBuilder
-              .append(h.name)
-              .append(": ")
+            appliedContentLength = appliedContentLength || h.name == `Content-Length`.name
+
+            stringBuilder.append(h.name).append(": ")
             appendSanitized(stringBuilder, h.value)
             stringBuilder.append(CRLF)
-            ()
           }
         }
 
-        if (
-          !appliedContentLength && req.body == EmptyBody && !NoPayloadMethods.contains(req.method)
-        ) {
+        def isEmptyBody = req.body eq EmptyBody
+        def isNoPayloadMethod = NoPayloadMethods.contains(req.method)
+        if (!appliedContentLength && isEmptyBody && !isNoPayloadMethod) {
           stringBuilder.append(zeroContentLengthRaw).append(CRLF)
           chunked = false
-        } else if (!chunked && !appliedContentLength && !NoPayloadMethods.contains(req.method)) {
+        } else if (!chunked && !appliedContentLength && !isNoPayloadMethod) {
```

Comment:
@danicheg wdyt? I studied the bytecode and looks like these are emitted as private static methods, which I think it's safe to assume that the JIT will inline.

Reply:
Can confirm – readability is better now :)